solr/core/src/java/org/apache/solr/search/SolrCoreParser.java

```diff
@@ -55,28 +60,40 @@ public void init(@SuppressWarnings("rawtypes") NamedList initArgs) {
       loader = req.getCore().getResourceLoader();
     }
 
-    @SuppressWarnings("unchecked")
     final Iterable<Map.Entry<String,Object>> args = initArgs;
     for (final Map.Entry<String,Object> entry : args) {
       final String queryName = entry.getKey();
       final String queryBuilderClassName = (String)entry.getValue();
 
       try {
-        final SolrQueryBuilder queryBuilder = loader.newInstance(
+        final SolrSpanQueryBuilder spanQueryBuilder = loader.newInstance(
             queryBuilderClassName,
-            SolrQueryBuilder.class,
+            SolrSpanQueryBuilder.class,
             null,
-            new Class[] {String.class, Analyzer.class, SolrQueryRequest.class, QueryBuilder.class},
-            new Object[] {defaultField, analyzer, req, this});
-        this.queryFactory.addBuilder(queryName, queryBuilder);
-      } catch (SolrException ex) {
+            new Class[] {String.class, Analyzer.class, SolrQueryRequest.class, QueryBuilder.class, SpanQueryBuilder.class},
+            new Object[] {defaultField, analyzer, req, this, this});
```

Comment:
@cpoerschke yes, I think that would work better as an API for the individual builders.